codehearts/remembear
# Add scheduler

## Description of changes

This adds a scheduler to fire off reminders in real-time. Nothing is immediately visible yet, that's where integrations will come in

## Reason for changes 

This runs the reminders in real-time, leaving integrations as the final piece of the puzzle to perform an action (or actions) when a reminder is scheduled to occur

## Test steps 

1. `cargo test`
1. Make sure you have no reminders and run the scheduler, expecting to see "Scheduler queue is empty"
       
       cargo run start
1. Add a user and reminder, the time doesn't matter

       cargo run user add donna
       cargo run reminder add roadhouse '{"mon":["21:00:00"]}' 1
1. Run the scheduler again and watch `remembear` just sit there doing nothing. Hit ctrl+c to stop it, integrations will make this step do something interesting in the future

       cargo run start
